File: padroesCriacionais/padrao-de-projeto-abstractFactory/php/src/Game.php
<?php

namespace Abstractfactory\Php;

class Game
{
    private ?iEnemies $enemies = null;
    private ?iMap $map = null;

    public function __construct(iAbstractGameFactory $abstractGameFactory)
    {
        $this->enemies = $abstractGameFactory->getEnemies();
        $this->map = $abstractGameFactory->getMap();
    }

    public function getEnemies(): String
    {
        return $this->enemies->makeEnemies();
    }

    public function getMap(): String
    {
        return $this->map->makeMap();
    }
}

File: padroesCriacionais/padrao-de-projeto-abstractFactory/java/src/main/java/abstractFactory/Game.java
package abstractFactory;

public class Game {

    public iEnemies enemies = null;
    public iMap map = null;

    public Game(iAbstractGameFactory abstractGameFactory) {
        this.enemies = abstractGameFactory.getEnemies();
        this.map = abstractGameFactory.getMap();
    }

    public String getEnemies() {
        return this.enemies.makeEnemies();
    }

    public String getMap() {
        return this.map.makeMap();
    }
}
